Lib/asyncio/locks.py

```diff
@@ -71,15 +71,13 @@ def __await__(self):
         yield from self.acquire()
```

Comment:
`with await lock` prevents converting `.acquire()` to async function. This is sad.

Reply:
`acquire` can be converted to `async def`, just do this:

```python
async def acquire(self):
    ...

def __await__(self):
    return self.acquire().__await__()
```

I'd also raise a `DeprecationWarning` in `Lock.__await__`.  The new idiomatic way is to write `async with lock`.  We can open a separate issue for this.

Reply:
The problem is that `__await__(self)` should return not awaitable for `self.acquire()` but an awaitable object with `__enter__` / `__exit__` for usage in statements like:
```
            lock = asyncio.Lock()
            with await lock as _lock:
                assert _lock is None
                assert lock.locked()
```

Reply:
Alright, you can do this then:

```python
async def _acquire_ctx(self):
    await self.acquire()
    return _ContextManager(self)

def __await__(self):
    return self. _acquire_ctx().__await__()
```

Reply:
(use two leading underscores, I want 0 chance of someone using `acquire_ctx` method)

Reply:
Done.